diff --git a/apps/image-editor/src/css/buttons.styl b/apps/image-editor/src/css/buttons.styl
index 339e4b30c..6b533c81e 100644
--- a/apps/image-editor/src/css/buttons.styl
+++ b/apps/image-editor/src/css/buttons.styl
@@ -22,6 +22,17 @@
   &.free .{prefix}-button.free svg > use.active
     display: block;
 
+/* HIHGLIGHT BUTTON */
+.tie-draw-highlights-select-button
+  &.line .{prefix}-button.line svg > use.normal,
+  &.free .{prefix}-button.free svg > use.normal
+    display: none;
+
+  &.line .{prefix}-button.line svg > use.active,
+  &.free .{prefix}-button.free svg > use.active
+    display: block;
+
+
 /* FLIP BUTTON */
 .tie-flip-button
   &.resetFlip .{prefix}-button.resetFlip,
diff --git a/apps/image-editor/src/css/submenu.styl b/apps/image-editor/src/css/submenu.styl
index af6898eba..d8c81ce8a 100644
--- a/apps/image-editor/src/css/submenu.styl
+++ b/apps/image-editor/src/css/submenu.styl
@@ -75,6 +75,7 @@
     .{prefix}-main.{prefix}-menu-mask .{prefix}-submenu > div.{prefix}-menu-mask,
     .{prefix}-main.{prefix}-menu-icon .{prefix}-submenu > div.{prefix}-menu-icon,
     .{prefix}-main.{prefix}-menu-draw .{prefix}-submenu > div.{prefix}-menu-draw,
+    .{prefix}-main.{prefix}-menu-highlights .{prefix}-submenu > div.{prefix}-menu-highlights,
     .{prefix}-main.{prefix}-menu-filter .{prefix}-submenu > div.{prefix}-menu-filter,
     .{prefix}-main.{prefix}-menu-zoom .{prefix}-submenu > div.{prefix}-menu-zoom
         display: table-cell;
@@ -87,6 +88,7 @@
     .{prefix}-main.{prefix}-menu-mask,
     .{prefix}-main.{prefix}-menu-icon,
     .{prefix}-main.{prefix}-menu-draw,
+    .{prefix}-main.{prefix}-menu-highlights,
     .{prefix}-main.{prefix}-menu-filter,
     .{prefix}-main.{prefix}-menu-zoom
       .{prefix}-submenu
diff --git a/apps/image-editor/src/js/action.js b/apps/image-editor/src/js/action.js
index f708da129..ecceb2625 100644
--- a/apps/image-editor/src/js/action.js
+++ b/apps/image-editor/src/js/action.js
@@ -20,6 +20,7 @@ export default {
       text: this._textAction(),
       mask: this._maskAction(),
       draw: this._drawAction(),
+      highlights: this._highlightsAction(),
       icon: this._iconAction(),
       filter: this._filterAction(),
       history: this._historyAction(),
@@ -261,6 +262,28 @@ export default {
     );
   },
 
+  /**
+   * HIGHLIGHT_LINE Action
+   * @returns {Object} actions for ui draw
+   * @private
+   */
+  _highlightsAction() {
+    return extend(
+      {
+        setHighlightsMode: (type, settings) => {
+          this.stopDrawingMode();
+          this.startDrawingMode('HIGHLIGHT_LINE', settings);
+        },
+        setColor: (color) => {
+          this.setBrush({
+            color,
+          });
+        },
+      },
+      this._commonAction()
+    );
+  },
+
   /**
    * Mask Action
    * @returns {Object} actions for ui mask
@@ -566,6 +589,11 @@ export default {
             this.ui.changeMenu('draw', false, false);
             this.ui.draw.changeStandbyMode();
           }
+        } else if (obj.type === 'highlight-line') {
+          if (this.ui.submenu !== 'highlights') {
+            this.ui.changeMenu('highlights', false, false);
+            this.ui.highlights.changeStandbyMode();
+          }
         } else if (['i-text', 'text'].indexOf(obj.type) > -1) {
           if (this.ui.submenu !== 'text') {
             this.ui.changeMenu('text', false, false);
diff --git a/apps/image-editor/src/js/component/highlightLine.js b/apps/image-editor/src/js/component/highlightLine.js
new file mode 100644
index 000000000..fe5752490
--- /dev/null
+++ b/apps/image-editor/src/js/component/highlightLine.js
@@ -0,0 +1,207 @@
+import { fabric } from 'fabric';
+import extend from 'tui-code-snippet/object/extend';
+import Component from '@/interface/component';
+import ArrowLine from '@/extension/arrowLine';
+import { eventNames, componentNames, fObjectOptions } from '@/consts';
+
+/**
+ * Highlights
+ * @class Highlights
+ * @param {Graphics} graphics - Graphics instance
+ * @extends {Component}
+ * @ignore
+ */
+class HighlightLine extends Component {
+  constructor(graphics) {
+    super(componentNames.HIGHLIGHT_LINE, graphics);
+
+    /**
+     * Brush width
+     * @type {number}
+     * @private
+     */
+    this._width = 12;
+
+    /**
+     * fabric.Color instance for brush color
+     * @type {fabric.Color}
+     * @private
+     */
+    this._oColor = new fabric.Color('rgba(0, 0, 0, 0.5)');
+
+    /**
+     * Listeners
+     * @type {object.<string, function>}
+     * @private
+     */
+    this._listeners = {
+      mousedown: this._onFabricMouseDown.bind(this),
+      mousemove: this._onFabricMouseMove.bind(this),
+      mouseup: this._onFabricMouseUp.bind(this),
+    };
+  }
+
+  /**
+   * Start drawing Highlights mode
+   * @param {{width: ?number, color: ?string}} [setting] - Brush width & color
+   */
+  setHeadOption(setting) {
+    const {
+      arrowType = {
+        head: null,
+        tail: null,
+      },
+    } = setting;
+
+    this._arrowType = arrowType;
+  }
+
+  /**
+   * Start drawing Highlights mode
+   * @param {{width: ?number, color: ?string}} [setting] - Brush width & color
+   */
+  start(setting = {}) {
+    const canvas = this.getCanvas();
+
+    canvas.defaultCursor = 'pointer';
+    canvas.selection = false;
+
+    this.setHeadOption(setting);
+    this.setBrush(setting);
+
+    canvas.forEachObject((obj) => {
+      obj.set({
+        evented: false,
+      });
+    });
+
+    canvas.on({
+      'mouse:down': this._listeners.mousedown,
+    });
+  }
+
+  /**
+   * Set brush
+   * @param {{width: ?number, color: ?string}} [setting] - Brush width & color
+   */
+  setBrush(setting) {
+    const brush = this.getCanvas().freeDrawingBrush;
+
+    setting = setting || {};
+    this._width = setting.width || this._width;
+
+    if (setting.color) {
+      this._oColor = new fabric.Color(setting.color);
+    }
+    brush.width = this._width;
+    brush.color = this._oColor.toRgba();
+  }
+
+  /**
+   * End drawing Highlights mode
+   */
+  end() {
+    const canvas = this.getCanvas();
+
+    canvas.defaultCursor = 'default';
+    canvas.selection = true;
+
+    canvas.forEachObject((obj) => {
+      obj.set({
+        evented: true,
+      });
+    });
+
+    canvas.off('mouse:down', this._listeners.mousedown);
+  }
+
+  /**
+   * Mousedown event handler in fabric canvas
+   * @param {{target: fabric.Object, e: MouseEvent}} fEvent - Fabric event object
+   * @private
+   */
+  _onFabricMouseDown(fEvent) {
+    const canvas = this.getCanvas();
+    const { x, y } = canvas.getPointer(fEvent.e);
+    const points = [x, y, x, y];
+
+    this._highlightLine = new ArrowLine(points, {
+      stroke: this._oColor.toRgba(),
+      strokeWidth: this._width,
+      arrowType: this._arrowType,
+      evented: false,
+    });
+    if (this._highlightLine.type === 'line') {
+      this._highlightLine.type = 'highlight-line';
+    }
+    this._highlightLine.set(fObjectOptions.SELECTION_STYLE);
+
+    canvas.add(this._highlightLine);
+
+    canvas.on({
+      'mouse:move': this._listeners.mousemove,
+      'mouse:up': this._listeners.mouseup,
+    });
+
+    this.fire(eventNames.ADD_OBJECT, this._createHighlightsEventObjectProperties());
+  }
+
+  /**
+   * Mousemove event handler in fabric canvas
+   * @param {{target: fabric.Object, e: MouseEvent}} fEvent - Fabric event object
+   * @private
+   */
+  _onFabricMouseMove(fEvent) {
+    const canvas = this.getCanvas();
+    const pointer = canvas.getPointer(fEvent.e);
+
+    this._highlightLine.set({
+      x2: pointer.x,
+      y2: pointer.y,
+    });
+
+    this._highlightLine.setCoords();
+
+    canvas.renderAll();
+  }
+
+  /**
+   * Mouseup event handler in fabric canvas
+   * @private
+   */
+  _onFabricMouseUp() {
+    const canvas = this.getCanvas();
+
+    this.fire(eventNames.OBJECT_ADDED, this._createHighlightsEventObjectProperties());
+
+    this._highlightLine = null;
+
+    canvas.off({
+      'mouse:move': this._listeners.mousemove,
+      'mouse:up': this._listeners.mouseup,
+    });
+  }
+
+  /**
+   * create Highlights event object properties
+   * @returns {Object} properties Highlights object
+   * @private
+   */
+  _createHighlightsEventObjectProperties() {
+    const params = this.graphics.createObjectProperties(this._highlightLine);
+    const { x1, x2, y1, y2 } = this._highlightLine;
+
+    return extend({}, params, {
+      startPosition: {
+        x: x1,
+        y: y1,
+      },
+      endPosition: {
+        x: x2,
+        y: y2,
+      },
+    });
+  }
+}
+
+export default HighlightLine;
diff --git a/apps/image-editor/src/js/consts.js b/apps/image-editor/src/js/consts.js
index 65ab07e10..4ea76c704 100644
--- a/apps/image-editor/src/js/consts.js
+++ b/apps/image-editor/src/js/consts.js
@@ -71,6 +71,8 @@ export const componentNames = keyMirror(
   'FLIP',
   'ROTATION',
   'FREE_DRAWING',
+  'HIGHLIGHT_DRAWING',
+  'HIGHLIGHT_LINE',
   'LINE',
   'TEXT',
   'ICON',
@@ -212,6 +214,8 @@ export const drawingModes = keyMirror(
   'CROPPER',
   'FREE_DRAWING',
   'LINE_DRAWING',
+  'HIGHLIGHT_DRAWING',
+  'HIGHLIGHT_LINE',
   'TEXT',
   'SHAPE',
   'ICON',
@@ -331,6 +335,12 @@ export const defaultDrawRangeValues = {
   value: 12,
 };
 
+export const defaultHighlightsRangeValues = {
+  min: 20,
+  max: 100,
+  value: 37,
+};
+
 export const defaultShapeStrokeValues = {
   realTimeEvent: true,
   min: 2,
diff --git a/apps/image-editor/src/js/drawingMode/highlightLine.js b/apps/image-editor/src/js/drawingMode/highlightLine.js
new file mode 100644
index 000000000..304f51fce
--- /dev/null
+++ b/apps/image-editor/src/js/drawingMode/highlightLine.js
@@ -0,0 +1,36 @@
+import DrawingMode from '@/interface/drawingMode';
+import { drawingModes, componentNames as components } from '@/consts';
+
+/**
+ * HighlightLineMode class
+ * @class
+ * @ignore
+ */
+class HighlightLineMode extends DrawingMode {
+  constructor() {
+    super(drawingModes.HIGHLIGHT_LINE);
+  }
+
+  /**
+   * start this drawing mode
+   * @param {Graphics} graphics - Graphics instance
+   * @param {{width: ?number, color: ?string}} [options] - Brush width & color
+   * @override
+   */
+  start(graphics, options) {
+    const lineDrawing = graphics.getComponent(components.HIGHLIGHT_LINE);
+    lineDrawing.start(options);
+  }
+
+  /**
+   * stop this drawing mode
+   * @param {Graphics} graphics - Graphics instance
+   * @override
+   */
+  end(graphics) {
+    const lineDrawing = graphics.getComponent(components.HIGHLIGHT_LINE);
+    lineDrawing.end();
+  }
+}
+
+export default HighlightLineMode;
diff --git a/apps/image-editor/src/js/graphics.js b/apps/image-editor/src/js/graphics.js
index 3b65096e8..45afac6f0 100644
--- a/apps/image-editor/src/js/graphics.js
+++ b/apps/image-editor/src/js/graphics.js
@@ -11,6 +11,7 @@ import Flip from '@/component/flip';
 import Rotation from '@/component/rotation';
 import FreeDrawing from '@/component/freeDrawing';
 import Line from '@/component/line';
+import HighlightLine from '@/component/highlightLine';
 import Text from '@/component/text';
 import Icon from '@/component/icon';
 import Filter from '@/component/filter';
@@ -19,6 +20,7 @@ import Zoom from '@/component/zoom';
 import CropperDrawingMode from '@/drawingMode/cropper';
 import FreeDrawingMode from '@/drawingMode/freeDrawing';
 import LineDrawingMode from '@/drawingMode/lineDrawing';
+import HighlightLineMode from '@/drawingMode/highlightLine';
 import ShapeDrawingMode from '@/drawingMode/shape';
 import TextDrawingMode from '@/drawingMode/text';
 import IconDrawingMode from '@/drawingMode/icon';
@@ -743,6 +745,10 @@ class Graphics {
       compName = components.LINE;
     }
 
+    if (drawingMode === drawingModes.HIGHLIGHT_LINE) {
+      compName = components.HIGHLIGHT_LINE;
+    }
+
     this.getComponent(compName).setBrush(option);
   }
 
@@ -827,7 +833,11 @@ class Graphics {
     const object = this.getObject(id);
     const clone = extend({}, props);
 
-    object.set(clone);
+    if (props.type && (props.type === 'line' || props.type === 'highlight-line')) {
+      this.deepCopyWithCircularRefs(clone, object);
+    } else {
+      object.set(clone);
+    }
 
     object.setCoords();
 
@@ -836,6 +846,31 @@ class Graphics {
     return clone;
   }
 
+  // eslint-disable-next-line complexity
+  deepCopyWithCircularRefs(source, target = {}, seen = new WeakMap()) {
+    if (seen.has(source)) {
+      return seen.get(source);
+    }
+
+    for (const key in source) {
+      if (key === 'canvas' || key === 'ctx' || key === 'ownMatrixCache') {
+        continue;
+      }
+      if (source.hasOwnProperty(key)) {
+        const value = source[key];
+        if (value && typeof value === 'object') {
+          target[key] = Array.isArray(value) ? [] : {};
+          seen.set(value, target[key]);
+          this.deepCopyWithCircularRefs(value, target[key], seen);
+        } else {
+          target[key] = value;
+        }
+      }
+    }
+
+    return target;
+  }
+
   /**
    * Get object properties corresponding key
    * @param {number} id - object id
@@ -989,6 +1024,7 @@ class Graphics {
     this._register(this._drawingModeMap, new CropperDrawingMode());
     this._register(this._drawingModeMap, new FreeDrawingMode());
     this._register(this._drawingModeMap, new LineDrawingMode());
+    this._register(this._drawingModeMap, new HighlightLineMode());
     this._register(this._drawingModeMap, new ShapeDrawingMode());
     this._register(this._drawingModeMap, new TextDrawingMode());
     this._register(this._drawingModeMap, new IconDrawingMode());
@@ -1006,6 +1042,7 @@ class Graphics {
     this._register(this._componentMap, new Flip(this));
     this._register(this._componentMap, new Rotation(this));
     this._register(this._componentMap, new FreeDrawing(this));
+    this._register(this._componentMap, new HighlightLine(this));
     this._register(this._componentMap, new Line(this));
     this._register(this._componentMap, new Text(this));
     this._register(this._componentMap, new Icon(this));
diff --git a/apps/image-editor/src/js/ui.js b/apps/image-editor/src/js/ui.js
index d67a18d8f..ad74a5503 100644
--- a/apps/image-editor/src/js/ui.js
+++ b/apps/image-editor/src/js/ui.js
@@ -22,6 +22,7 @@ import Text from '@/ui/text';
 import Mask from '@/ui/mask';
 import Icon from '@/ui/icon';
 import Draw from '@/ui/draw';
+import Highlights from '@/ui/highlights';
 import Filter from '@/ui/filter';
 import History from '@/ui/history';
 import Locale from '@/ui/locale/locale';
@@ -36,6 +37,7 @@ const SUB_UI_COMPONENT = {
   Mask,
   Icon,
   Draw,
+  Highlights,
   Filter,
 };
 
@@ -255,6 +257,7 @@ class Ui {
           'flip',
           'rotate',
           'draw',
+          'highlights',
           'shape',
           'icon',
           'text',
diff --git a/apps/image-editor/src/js/ui/highlights.js b/apps/image-editor/src/js/ui/highlights.js
new file mode 100644
index 000000000..108139065
--- /dev/null
+++ b/apps/image-editor/src/js/ui/highlights.js
@@ -0,0 +1,198 @@
+import Colorpicker from '@/ui/tools/colorpicker';
+import Range from '@/ui/tools/range';
+import Submenu from '@/ui/submenuBase';
+import templateHtml from '@/ui/template/submenu/highlights';
+import { assignmentForDestroy, getRgb } from '@/util';
+import { defaultHighlightsRangeValues, eventNames, selectorNames } from '@/consts';
+
+const HIGHLIGHTS_OPACITY = 0.35;
+
+/**
+ * Highlights ui class
+ * @class
+ * @ignore
+ */
+class Highlights extends Submenu {
+  constructor(subMenuElement, { locale, makeSvgIcon, menuBarPosition, usageStatistics }) {
+    super(subMenuElement, {
+      locale,
+      name: 'highlights',
+      makeSvgIcon,
+      menuBarPosition,
+      templateHtml,
+      usageStatistics,
+    });
+
+    this._els = {
+      lineSelectButton: this.selector('.tie-highlights-line-select-button'),
+      highlightsColorPicker: new Colorpicker(this.selector('.tie-highlights-color'), {
+        defaultColor: '#00a9ff',
+        toggleDirection: this.toggleDirection,
+        usageStatistics: this.usageStatistics,
+      }),
+      highlightsRange: new Range(
+        {
+          slider: this.selector('.tie-highlights-range'),
+          input: this.selector('.tie-highlights-range-value'),
+        },
+        defaultHighlightsRangeValues
+      ),
+    };
+
+    this.type = null;
+    this.color = this._els.highlightsColorPicker.color;
+    this.width = this._els.highlightsRange.value;
+
+    this.colorPickerInputBox = this._els.highlightsColorPicker.colorpickerElement.querySelector(
+      selectorNames.COLOR_PICKER_INPUT_BOX
+    );
+  }
+
+  /**
+   * Destroys the instance.
+   */
+  destroy() {
+    this._removeEvent();
+    this._els.highlightsColorPicker.destroy();
+    this._els.highlightsRange.destroy();
+
+    assignmentForDestroy(this);
+  }
+
+  /**
+   * Add event for Highlights
+   * @param {Object} actions - actions for crop
+   *   @param {Function} actions.setHighlightsMode - set Highlights mode
+   */
+  addEvent(actions) {
+    this.eventHandler.changeHighlightsType = this._changeHighlightsType.bind(this);
+
+    this.actions = actions;
+    this._els.lineSelectButton.addEventListener('click', this.eventHandler.changeHighlightsType);
+    this._els.highlightsColorPicker.on('change', this._changeHighlightsColor.bind(this));
+    this._els.highlightsRange.on('change', this._changeHighlightsRange.bind(this));
+
+    this.colorPickerInputBox.addEventListener(
+      eventNames.FOCUS,
+      this._onStartEditingInputBox.bind(this)
+    );
+    this.colorPickerInputBox.addEventListener(
+      eventNames.BLUR,
+      this._onStopEditingInputBox.bind(this)
+    );
+  }
+
+  /**
+   * Remove event
+   * @private
+   */
+  _removeEvent() {
+    this._els.lineSelectButton.removeEventListener('click', this.eventHandler.changeHighlightsType);
+    this._els.highlightsColorPicker.off();
+    this._els.highlightsRange.off();
+
+    this.colorPickerInputBox.removeEventListener(
+      eventNames.FOCUS,
+      this._onStartEditingInputBox.bind(this)
+    );
+    this.colorPickerInputBox.removeEventListener(
+      eventNames.BLUR,
+      this._onStopEditingInputBox.bind(this)
+    );
+  }
+
+  /**
+   * set Highlights mode - action runner
+   */
+  setDrawMode() {
+    this.actions.setHighlightsMode(this.type, {
+      width: this.width,
+      color: getRgb(this.color, HIGHLIGHTS_OPACITY),
+    });
+  }
+
+  /**
+   * Returns the menu to its default state.
+   */
+  changeStandbyMode() {
+    this.type = null;
+    this.actions.stopDrawingMode();
+    this.actions.changeSelectableAll(true);
+    this._els.lineSelectButton.classList.remove('free');
+    this._els.lineSelectButton.classList.remove('line');
+  }
+
+  /**
+   * Executed when the menu starts.
+   */
+  changeStartMode() {
+    this.type = 'highlight-draw';
+    this._els.lineSelectButton.classList.add('free');
+    this.setDrawMode();
+  }
+
+  /**
+   * Change Highlights type event
+   * @param {object} event - line select event
+   * @private
+   */
+  _changeHighlightsType(event) {
+    const button = event.target.closest('.tui-image-editor-button');
+    if (button) {
+      const highlightType = this.getButtonType(button, ['free', 'line']);
+      this.actions.discardSelection();
+      let lineType = '';
+      if (highlightType === 'free') lineType = 'highlight-drawing';
+      if (highlightType === 'line') lineType = 'highlight-line';
+
+      if (this.type === lineType) {
+        this.changeStandbyMode();
+
+        return;
+      }
+
+      this.changeStandbyMode();
+      this.type = lineType;
+      this._els.lineSelectButton.classList.add(lineType);
+      this.setDrawMode();
+    }
+  }
+
+  _setHighlightsType() {
+    this.actions.discardSelection();
+    this.changeStandbyMode();
+    this.type = 'highlight-line';
+    this._els.lineSelectButton.classList.add('highlight-line');
+    this.setDrawMode();
+  }
+
+  /**
+   * Change Highlightsing color
+   * @param {string} color - select Highlightsing color
+   * @private
+   */
+  _changeHighlightsColor(color) {
+    this.color = color || 'transparent';
+    if (!this.type) {
+      this.changeStartMode();
+    } else {
+      this.setDrawMode();
+    }
+  }
+
+  /**
+   * Change Highlightsing Range
+   * @param {number} value - select Highlightsing range
+   * @private
+   */
+  _changeHighlightsRange(value) {
+    this.width = value;
+    if (!this.type) {
+      this.changeStartMode();
+    } else {
+      this.setDrawMode();
+    }
+  }
+}
+
+export default Highlights;
diff --git a/apps/image-editor/src/js/ui/template/style.js b/apps/image-editor/src/js/ui/template/style.js
index 3eb70e1fc..7db76751d 100644
--- a/apps/image-editor/src/js/ui/template/style.js
+++ b/apps/image-editor/src/js/ui/template/style.js
@@ -32,6 +32,7 @@ export default ({
     .tie-icon-add-button.icon-bubble .tui-image-editor-button[data-icontype="icon-bubble"] label,
     .tie-draw-line-select-button.line .tui-image-editor-button.line label,
     .tie-draw-line-select-button.free .tui-image-editor-button.free label,
+    .tie-highlights-line-select-button.line .tui-image-editor-button.line label,
     .tie-flip-button.flipX .tui-image-editor-button.flipX label,
     .tie-flip-button.flipY .tui-image-editor-button.flipY label,
     .tie-flip-button.resetFlip .tui-image-editor-button.resetFlip label,
diff --git a/apps/image-editor/src/js/ui/template/submenu/highlights.js b/apps/image-editor/src/js/ui/template/submenu/highlights.js
new file mode 100644
index 000000000..b44adb1e5
--- /dev/null
+++ b/apps/image-editor/src/js/ui/template/submenu/highlights.js
@@ -0,0 +1,34 @@
+/**
+ * @param {Object} submenuInfo - submenu info for make template
+ *   @param {Locale} locale - Translate text
+ *   @param {Function} makeSvgIcon - svg icon generator
+ * @returns {string}
+ */
+export default ({ locale, makeSvgIcon }) => `
+    <ul class="tui-image-editor-submenu-item">
+        <li class="tie-highlights-line-select-button" style="visibility:hidden">
+            <div class="tui-image-editor-button line">
+                <div>
+                    ${makeSvgIcon(['normal', 'active'], 'draw-line', true)}
+                </div>
+                <label>
+                    ${locale.localize('Highlight Line')}
+                </label>
+            </div>
+        </li>
+        <li class="tui-image-editor-partition" style="visibility:hidden">
+            <div></div>
+        </li>
+        <li>
+            <div class="tie-highlights-color" title="${locale.localize('Color')}"></div>
+        </li>
+        <li class="tui-image-editor-partition only-left-right">
+            <div></div>
+        </li>
+        <li class="tui-image-editor-newline tui-image-editor-range-wrap">
+            <label class="range">${locale.localize('Range')}</label>
+            <div class="tie-highlights-range"></div>
+            <input class="tie-highlights-range-value tui-image-editor-range-value" value="0" />
+        </li>
+    </ul>
+`;
diff --git a/apps/image-editor/src/svg/default.svg b/apps/image-editor/src/svg/default.svg
index ef483b47c..2cd067fdc 100644
--- a/apps/image-editor/src/svg/default.svg
+++ b/apps/image-editor/src/svg/default.svg
@@ -330,6 +330,22 @@
         <path fill="inherit" fill-rule="nonzero" d="M12.5 1.5c1.325 0 2.41 1.032 2.495 2.336L15 4v7.22h-1V4c0-.828-.672-1.5-1.5-1.5-.78 0-1.42.595-1.493 1.356L11 4v7.22h-1V4c0-1.38 1.12-2.5 2.5-2.5z" transform="translate(-297 -290) translate(297 290)"/>
         <path fill="inherit" fill-rule="nonzero" d="M16.5 3.5c1.325 0 2.41 1.032 2.495 2.336L19 6v6.3h-1V6c0-.828-.672-1.5-1.5-1.5-.78 0-1.42.595-1.493 1.356L15 6v2.44h-1V6c0-1.38 1.12-2.5 2.5-2.5z" transform="translate(-297 -290) translate(297 290)"/>
     </g>
+</symbol>
+ <symbol id="ic-hideunhide" viewBox="0 0 32 32">
+        <rect style="stroke: rgb(0, 0, 0); fill: rgb(72, 252, 91);" y="0.548" width="31.014" height="30.904" rx="2"
+            ry="2" x="0.548" />
+        <rect style="stroke: rgb(0, 0, 0);" width="10" height="24" rx="1" ry="1" x="18" y="4" />
+</symbol>
+<symbol id="ic-hideunhide-off" viewBox="0 0 32 32">
+    <rect style="stroke: rgb(0, 0, 0); fill: rgb(199, 199, 199);" y="0.548" width="31.014" height="30.904"
+        rx="2" ry="2" x="0.548" />
+    <rect style="stroke: rgb(0, 0, 0);" width="10" height="24" rx="1" ry="1" x="4" y="4" />
+</symbol>
+<symbol id="ic-highlights" viewBox="0 0 24 24">
+ <g fill="none" transform="matrix(0.9999929666519164, 0.00370400003157556, -0.00370400003157556, 0.9999929666519164, 29.490911483764645, -7.770881175994873)">
+    <path fill="#434343" d="M -19.259 17.087 L -15.369 20.977 L -16.077 21.683 L -19.966 17.794 L -19.259 17.087 Z"/>
+  </g>
+  <path d="M 0.917 23.974 L 0.916 21.168 L 23.786 21.168 L 23.787 23.974 L 0.917 23.974 Z M 16.45 11.006 L 12.375 7.083 L 7.015 12.141 L 11.06 16.132 L 16.45 11.006 Z M 13.182 6.408 L 17.176 10.266 L 22.658 5.051 L 18.583 1.35 L 13.182 6.408 Z M 11.581 5.837 L 17.755 11.693 L 12.352 16.843 C 11.895 17.278 11.357 17.494 10.738 17.494 C 10.118 17.494 9.58 17.278 9.122 16.843 L 6.405 18.799 L 0.688 18.799 L 6.235 14.079 C 5.778 13.645 5.541 13.125 5.52 12.519 C 5.501 11.915 5.721 11.395 6.179 10.961 L 11.581 5.837 Z M 11.581 5.837 L 16.927 0.768 C 17.383 0.334 17.922 0.118 18.542 0.118 C 19.161 0.118 19.7 0.334 20.157 0.768 L 23.129 3.559 C 23.586 3.994 23.816 4.504 23.816 5.092 C 23.816 5.679 23.586 6.189 23.129 6.623 L 17.755 11.693 L 11.581 5.837 Z" style="fill: rgb(67, 67, 67); transform-origin: 12.252px 12.047px;" transform="matrix(0.99999499321, 0.003291000146, -0.003291000146, 0.99999499321, -3.427e-9, -2.505e-9)"/>
 </symbol>
 </defs>
 </svg>
diff --git a/apps/image-editor/src/svg/icon-a/ic-highlights.svg b/apps/image-editor/src/svg/icon-a/ic-highlights.svg
new file mode 100644
index 000000000..9dbf27c58
--- /dev/null
+++ b/apps/image-editor/src/svg/icon-a/ic-highlights.svg
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+ <g fill="none" transform="matrix(0.9999929666519164, 0.00370400003157556, -0.00370400003157556, 0.9999929666519164, 29.490911483764645, -7.770881175994873)">
+    <path fill="#434343" d="M -19.259 17.087 L -15.369 20.977 L -16.077 21.683 L -19.966 17.794 L -19.259 17.087 Z"/>
+  </g>
+  <path d="M 0.917 23.974 L 0.916 21.168 L 23.786 21.168 L 23.787 23.974 L 0.917 23.974 Z M 16.45 11.006 L 12.375 7.083 L 7.015 12.141 L 11.06 16.132 L 16.45 11.006 Z M 13.182 6.408 L 17.176 10.266 L 22.658 5.051 L 18.583 1.35 L 13.182 6.408 Z M 11.581 5.837 L 17.755 11.693 L 12.352 16.843 C 11.895 17.278 11.357 17.494 10.738 17.494 C 10.118 17.494 9.58 17.278 9.122 16.843 L 6.405 18.799 L 0.688 18.799 L 6.235 14.079 C 5.778 13.645 5.541 13.125 5.52 12.519 C 5.501 11.915 5.721 11.395 6.179 10.961 L 11.581 5.837 Z M 11.581 5.837 L 16.927 0.768 C 17.383 0.334 17.922 0.118 18.542 0.118 C 19.161 0.118 19.7 0.334 20.157 0.768 L 23.129 3.559 C 23.586 3.994 23.816 4.504 23.816 5.092 C 23.816 5.679 23.586 6.189 23.129 6.623 L 17.755 11.693 L 11.581 5.837 Z" style="fill: rgb(67, 67, 67); transform-origin: 12.252px 12.047px;" transform="matrix(0.99999499321, 0.003291000146, -0.003291000146, 0.99999499321, -3.427e-9, -2.505e-9)"/>
+
+</svg>
\ No newline at end of file
diff --git a/apps/image-editor/src/svg/icon-b/ic-highlights.svg b/apps/image-editor/src/svg/icon-b/ic-highlights.svg
new file mode 100644
index 000000000..9dbf27c58
--- /dev/null
+++ b/apps/image-editor/src/svg/icon-b/ic-highlights.svg
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+ <g fill="none" transform="matrix(0.9999929666519164, 0.00370400003157556, -0.00370400003157556, 0.9999929666519164, 29.490911483764645, -7.770881175994873)">
+    <path fill="#434343" d="M -19.259 17.087 L -15.369 20.977 L -16.077 21.683 L -19.966 17.794 L -19.259 17.087 Z"/>
+  </g>
+  <path d="M 0.917 23.974 L 0.916 21.168 L 23.786 21.168 L 23.787 23.974 L 0.917 23.974 Z M 16.45 11.006 L 12.375 7.083 L 7.015 12.141 L 11.06 16.132 L 16.45 11.006 Z M 13.182 6.408 L 17.176 10.266 L 22.658 5.051 L 18.583 1.35 L 13.182 6.408 Z M 11.581 5.837 L 17.755 11.693 L 12.352 16.843 C 11.895 17.278 11.357 17.494 10.738 17.494 C 10.118 17.494 9.58 17.278 9.122 16.843 L 6.405 18.799 L 0.688 18.799 L 6.235 14.079 C 5.778 13.645 5.541 13.125 5.52 12.519 C 5.501 11.915 5.721 11.395 6.179 10.961 L 11.581 5.837 Z M 11.581 5.837 L 16.927 0.768 C 17.383 0.334 17.922 0.118 18.542 0.118 C 19.161 0.118 19.7 0.334 20.157 0.768 L 23.129 3.559 C 23.586 3.994 23.816 4.504 23.816 5.092 C 23.816 5.679 23.586 6.189 23.129 6.623 L 17.755 11.693 L 11.581 5.837 Z" style="fill: rgb(67, 67, 67); transform-origin: 12.252px 12.047px;" transform="matrix(0.99999499321, 0.003291000146, -0.003291000146, 0.99999499321, -3.427e-9, -2.505e-9)"/>
+
+</svg>
\ No newline at end of file
diff --git a/apps/image-editor/src/svg/icon-c/ic-highlights.svg b/apps/image-editor/src/svg/icon-c/ic-highlights.svg
new file mode 100644
index 000000000..9dbf27c58
--- /dev/null
+++ b/apps/image-editor/src/svg/icon-c/ic-highlights.svg
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+ <g fill="none" transform="matrix(0.9999929666519164, 0.00370400003157556, -0.00370400003157556, 0.9999929666519164, 29.490911483764645, -7.770881175994873)">
+    <path fill="#434343" d="M -19.259 17.087 L -15.369 20.977 L -16.077 21.683 L -19.966 17.794 L -19.259 17.087 Z"/>
+  </g>
+  <path d="M 0.917 23.974 L 0.916 21.168 L 23.786 21.168 L 23.787 23.974 L 0.917 23.974 Z M 16.45 11.006 L 12.375 7.083 L 7.015 12.141 L 11.06 16.132 L 16.45 11.006 Z M 13.182 6.408 L 17.176 10.266 L 22.658 5.051 L 18.583 1.35 L 13.182 6.408 Z M 11.581 5.837 L 17.755 11.693 L 12.352 16.843 C 11.895 17.278 11.357 17.494 10.738 17.494 C 10.118 17.494 9.58 17.278 9.122 16.843 L 6.405 18.799 L 0.688 18.799 L 6.235 14.079 C 5.778 13.645 5.541 13.125 5.52 12.519 C 5.501 11.915 5.721 11.395 6.179 10.961 L 11.581 5.837 Z M 11.581 5.837 L 16.927 0.768 C 17.383 0.334 17.922 0.118 18.542 0.118 C 19.161 0.118 19.7 0.334 20.157 0.768 L 23.129 3.559 C 23.586 3.994 23.816 4.504 23.816 5.092 C 23.816 5.679 23.586 6.189 23.129 6.623 L 17.755 11.693 L 11.581 5.837 Z" style="fill: rgb(67, 67, 67); transform-origin: 12.252px 12.047px;" transform="matrix(0.99999499321, 0.003291000146, -0.003291000146, 0.99999499321, -3.427e-9, -2.505e-9)"/>
+
+</svg>
\ No newline at end of file
diff --git a/apps/image-editor/src/svg/icon-d/ic-highlights.svg b/apps/image-editor/src/svg/icon-d/ic-highlights.svg
new file mode 100644
index 000000000..9dbf27c58
--- /dev/null
+++ b/apps/image-editor/src/svg/icon-d/ic-highlights.svg
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+ <g fill="none" transform="matrix(0.9999929666519164, 0.00370400003157556, -0.00370400003157556, 0.9999929666519164, 29.490911483764645, -7.770881175994873)">
+    <path fill="#434343" d="M -19.259 17.087 L -15.369 20.977 L -16.077 21.683 L -19.966 17.794 L -19.259 17.087 Z"/>
+  </g>
+  <path d="M 0.917 23.974 L 0.916 21.168 L 23.786 21.168 L 23.787 23.974 L 0.917 23.974 Z M 16.45 11.006 L 12.375 7.083 L 7.015 12.141 L 11.06 16.132 L 16.45 11.006 Z M 13.182 6.408 L 17.176 10.266 L 22.658 5.051 L 18.583 1.35 L 13.182 6.408 Z M 11.581 5.837 L 17.755 11.693 L 12.352 16.843 C 11.895 17.278 11.357 17.494 10.738 17.494 C 10.118 17.494 9.58 17.278 9.122 16.843 L 6.405 18.799 L 0.688 18.799 L 6.235 14.079 C 5.778 13.645 5.541 13.125 5.52 12.519 C 5.501 11.915 5.721 11.395 6.179 10.961 L 11.581 5.837 Z M 11.581 5.837 L 16.927 0.768 C 17.383 0.334 17.922 0.118 18.542 0.118 C 19.161 0.118 19.7 0.334 20.157 0.768 L 23.129 3.559 C 23.586 3.994 23.816 4.504 23.816 5.092 C 23.816 5.679 23.586 6.189 23.129 6.623 L 17.755 11.693 L 11.581 5.837 Z" style="fill: rgb(67, 67, 67); transform-origin: 12.252px 12.047px;" transform="matrix(0.99999499321, 0.003291000146, -0.003291000146, 0.99999499321, -3.427e-9, -2.505e-9)"/>
+
+</svg>
\ No newline at end of file

